### PR TITLE
feat: Add food price index generator and figure chart generator funct…

### LINKED ISSUE #40: 4. box function

### LINKED ISSUE #45: 4.1 Food Price Index data function
- Build the food price index data function to generate food price index data based on the feed raw price data

### PR DESCRIPTION
**Pull request by tonyshumlh**
@celestezhao please note that food price index generator function that you might ned
@JohnShiuMK @srfrew please note the figure chart generator function for frontend code and backend code design

### CODE REVIEW

**Comment by celestezhao** [comment]
@tonyshumlh 

Hi Tony, your code is awesome! 

Just one thing - when I use your generate_food_price_index_data() function, seems there is a mismatching between the desired `widget_date_range` and the actual date scope in `price_index_data` df:
fdi_cal
Here I make the widget_date_range start from year 2010, and the starting year should be 2011 in returned price_index_data as our data starts from 2011. However, price_index_data returns 2014.

I'm thinking if this problem is due to `food_price_index_dict` - if the start date we expect does not have data for all the commodities defined in food_price_index_dict, then after executing the following code, records of these dates will be filtered out:
```
price_index_data = price_index_data[
    price_index_data.index_weight == len(food_price_index_dict)
].drop(columns=["index_weight"])
```
In the final selection of price_index_data, you require that records for each date must include all three commodities (Rice, Radish, Sugar). If a date is missing at least one of these commodities, then the record for that date will not be included in the final DataFrame.

Please lmk if you have some thoughts. Thanks again for calculating food price index, this has helped me a lot!

**Comment by srfrew** [comment]
This code functions well given our testing dataset retrieved using the fetch_data functions in #49 

Will complete review shortly

File: src/plotting.py
Comment (srfrew) [inline src/plotting.py:1]: This code and output is great! Thank you for such awesome work. 

Tested locally with fetch_data functions and functions as intended 
Unintended behaviour occurs if the date range is outside the provided data date range

**Review by srfrew — APPROVED** [review]
Code functions locally with fetch_data functions! Great work

**Comment by srfrew** [comment]
Local testing during review: 

```python
from fetch_data import fetch_country_data, fetch_country_index
from plotting import generate_figure_chart, generate_food_price_index_data
import pandas as pd
import altair as alt
alt.data_transformers.enable('vegafusion')

df = fetch_country_data("Japan", country_index_df = fetch_country_index())
figure_chart = generate_figure_chart(df, ('2018-01-01', '2020-09-15'), ["Osaka", "Tokyo"], ["Rice", "Radish"])
figure_chart[0]
```

**Comment by srfrew** [comment]
Merged YML conflict from #47. Code review complete, committing to main.

**Comment by srfrew** [comment]
Again awesome work Tony!

**Pull request merged**